src/app/modules/sample-providers/components/submission-form/submission-form.component.ts

```diff
@@ -1,22 +1,74 @@
-import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import {
+  Component,
+  OnInit,
+  Output,
+  EventEmitter,
+  Input,
+  ElementRef,
+  ViewChild,
+  HostListener,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+} from '@angular/core';
 import {
   AbstractControl,
   FormBuilder,
   FormGroup,
   Validators,
+  NG_VALUE_ACCESSOR,
+  ControlValueAccessor,
 } from '@angular/forms';
 
 @Component({
   selector: 'app-submission-form',
   templateUrl: './submission-form.component.html',
   styleUrls: ['./submission-form.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: SubmissionFormComponent,
+      multi: true,
+    },
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class SubmissionFormComponent implements OnInit {
+export class SubmissionFormComponent implements OnInit, ControlValueAccessor {
   form: FormGroup;
 
+  f: File; // demo file
+
+  @ViewChild('p', { read: ElementRef }) p: ElementRef;
+
   @Output()
   submitForm = new EventEmitter<any>();
 
+  @Input()
+  onChange = (f: File) => {};
+
+  @HostListener('change', ['$event.target.files'])
```

Comment:
This is going to create buggy behavior. Not only does this fire when a file is added, it will also fire when changes happen in any form control. Since a textarea, for example, does not have a concept of `files` it will fire a runtime error. 

One way to fix this issue is to create a new component called something like `audio-player-control` that could emit the value of the file back up to the form. This was the `HostListener` would only be listening to changes on the native `audio` tag.